synfig-studio/src/gui/instance.cpp

```diff
@@ -288,10 +295,48 @@ studio::Instance::run_plugin(std::string plugin_path)
 
 		canvas=0;
 
-		App::open_as(launcher.get_result_path(),launcher.get_original_path());
-
-
+		//
+		// QUICKHACK:
+		// Now dances with files :( (see https://github.com/synfig/synfig/issues/949)
+		//
+		// TODO: Replace copy operations with giomm???
+		//
+		// create a backup of original saved file
+		String backup_filename;
+		struct stat buf;
+		do {
```

Comment:
Why not to use [Glib::mkstemp(std::string& filename_template)](https://developer.gnome.org/glibmm/2.60/group__FileUtils.html) here?

Reply:
@ice0 I will try, thank you! ^__^

Reply:
This wouldn't work, because it returns `file descriptor` (int), while I need `fstream`.
It is possible to get fstream from file descriptor, but it is non-standard - https://stackoverflow.com/questions/2746168/how-to-construct-a-c-fstream-from-a-posix-file-descriptor